roachtest/sqlsmith: index out of range in geomfn.distanceLineStringInternal

roachtest.sqlsmith/setup=empty/setting=no-mutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2993935&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2993935&tab=artifacts#/sqlsmith/setup=empty/setting=no-mutations) on master @ [eecead5c3aa93b5718a57c47970d67ee3035e794](https://github.com/cockroachdb/cockroach/commits/eecead5c3aa93b5718a57c47970d67ee3035e794):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=empty/setting=no-mutations/run_1
	sqlsmith.go:215,sqlsmith.go:251,test_runner.go:792: error: pq: internal error: runtime error: index out of range [0] with length 0
		stmt:
		SELECT
			current_time(6958237490021249646:::INT8::INT8):::TIME::TIME AS col_41961
		FROM
			(VALUES (NULL), (e'\x1e\x11pP\x10':::STRING), (e'\x03"c!2\x02Y\x05N':::STRING)) AS tab_24266 (col_41956)
			JOIN (
					VALUES
						('10:24:49.164014-04:08:00':::TIMETZ, NULL, '15:22:12.646057':::TIME),
						(COALESCE(NULL, '23:28:39.645999-09:57:00':::TIMETZ), '_':::STRING, NULL),
						('21:54:46.546552-12:05:00':::TIMETZ, e'\'':::STRING, '00:00:00':::TIME),
						('18:47:07.173222-06:47:00':::TIMETZ, NULL, NULL),
						(
							timezone('X':::STRING::STRING, '08:50:56.168512+07:49:00':::TIMETZ::TIMETZ)::TIMETZ,
							NULL,
							'05:57:36.094826':::TIME
						),
						('03:20:05.746705+09:05:00':::TIMETZ, 'R':::STRING, '00:41:32.584509':::TIME)
				)
					AS tab_24267 (col_41957, col_41958, col_41959) ON true,
			(
				VALUES
					(
						'01060000C00900000001030000C00100000004000000B6570A686942F3C1A8CA99807DCAD041DE7C54515D49E8C1F03028280B010042E06294F8B14CC141C4528639DAC6E0C10A74B80C1CAFE4C158483964F942D5414CE1867028E4FD41DA82F05AD10A0242900F7C240D3C02C234D9673150CFE7C1B6570A686942F3C1A8CA99807DCAD041DE7C54515D49E8C1F03028280B01004201030000C0010000000A000000A26D828E9667FEC11E3617DA4EFDFBC1EC8E32B75E4BDBC1CEB3246C6B0CE7C1535B746EAB17F7C1BE3EB5855D63FEC1C01023F1B2CAA541B8E86678636B0242BE0B306FC211F9410EC2364EBD5CEFC11ED8F746056B01423AA02F7B4C0402421C36B134DF9DFF4164B696B8A241DDC1E2839E9DB638E9C1F2B94368288DF9C1604D2C13C989F841A8C4B5ECB777FC41044C75149B78ED41541C1AB5437AF14102DD4B5F3F0CF34154FACC6A83B6FF4192B774EF89EFE4C1707BC9BA99A2D0C1308A2416E320C6410CBCC64EEC6AE84120F03A83388D0242B42C47D094D3FF417E22F98EFA49E3C1B8BEDEAB70CD01421C8238E1218BFB4180CB865D1134FD416077FC620375D2C130C3DA1775A5DB416038F9C410ABDA41A044FA188971C6C1A26D828E9667FEC11E3617DA4EFDFBC1EC8E32B75E4BDBC1CEB3246C6B0CE7C101030000C00100000004000000CC2AADF64BCEFA41B0ED96A9ED06CEC1202ACE926474E241A2E5A16347ED00C2E05956E8AF44FB41F8DCBF811E19DF412C3B6C6D4111F3C18A272AAEE516F641209FFAD96BE900C24C88AEB32E09E841CD1BD17B9DA4F1C1403FD31FECC8B8C1CC2AADF64BCEFA41B0ED96A9ED06CEC1202ACE926474E241A2E5A16347ED00C201030000C0010000000A000000B3503AAD377501C22A06D8883B65FCC1EC064E09F3D1F84120476825894CD141E8505A9F9B7EDDC12D2EBB93850C01C2303483A800CCE44110D5C33BFC7AFA41F0611D4AB7F9F941861CDDA06105FAC125D7B273AA40FBC18CB7A3C6FD7AFA415CA939DDADA100423885412D1EF6DCC10C37C12C79E8E1410ED6B6D6B64DFBC10C8A6AA18FCAF041148D1BB13E93E041643C1A0EF4C2EA41DC57C28CD6CEF1C1A009467FDAD1DE41A0DC5FDAC64AF8418C2ACF239E0DF0419894963EB8F4F141D2D523CF88ADF8C19464E8BF3807E941E472F1D1671FFF410AFBAEB170ECEDC1988AF7E6F92BFBC1309E12E3539DE9414C9AF0EDE5EFFE41407CA0D82CA8E2416E07E7E728B8EBC100D45809AB706B415003B04D037FF241AE388240FE94EBC1B3503AAD377501C22A06D8883B65FCC1EC064E09F3D1F84120476825894CD14101030000C0010000000C000000014FD08BBE55F2C1FACBB99FB971E5C1F741E2AD2794F2C1FD8CEDD46AB201C2F6717F56BF47FBC1900C980C40DCFCC1BCFBDF0161460242E6EA1C83DF42E2C1B4C0AD9142F9EAC1FC1B3A268FD4D9C1C4B528E7FE15F9C1FE10672027DEF0412ED86F685CDBEAC12C0C68A67936D9C1B623601858DCE6C150AE7162CDE6D0C1E07376A5A10EC4C1EC2D8EB07F57ECC1346315D55A08F141D23A2269A86BFAC1D8ACC93842EDF741816A458C100101C2DAB59AF807B0F0411CFD35CE7475F241208C42097008B5C1CEA74BC56380F341B6A6452BB3CFF9419EAA1F09D67FF8C1E04C0884BF21D5C1CADBA0D9B57CF141ED4F62A47E5D02C281A978E2125CF1C127CA6A11135AF3C1DC368F10CADEED41D063F908935F01C2FCE1F32BCEB2E241166F4FA7A64FFDC1186892F64DF1F241005891F1435D6EC174525E0E36BCE94114148E25AD35FCC148F5BFAC2238E2413E0E43090CA5EDC1B3F9D25EDF13F7C1014FD08BBE55F2C1FACBB99FB971E5C1F741E2AD2794F2C1FD8CEDD46AB201C201030000C0010000000B00000016FE88D632F7F5C1FEA56B2CFAABFEC180A1951D8825A2C1EE7650BFB026FDC1803D78D86CF99A413A6F54EE6A5FE8C1F02A0B4730AFFD410CC4B6B229F8F4C1181A2BC61366D3C11C139EE79866F7C1CCDC4CFAB3CEE1C1803F7304E4C2E6410C190322C5C0F94177ABA2C555CF01C20A18821F8099E1C18CA50D73E4FAFC41F83242077EB8FE41CEACE3019C7CF0C1800CF0D7F72CEB4134B2F66AAA4EF241809A46B889130142C666A9B24879024271B44E05BC39F0C186A0C2878068FFC1688B016DEED2FA4100A59A777F8FFE41AC57D7677968F9C136147E2189A3004240513F21039FDB41500AD6DFAB0CD641E87F2848A083F74138B57F2B08DDEA41B2314F7F8A69E2C1183A738D99DA01422053B79C797DB741C076A95E6981CFC160A627E8B806E8C1403647422745EC41E8BD7B28A9DAFC413463A808CAC2EFC116FE88D632F7F5C1FEA56B2CFAABFEC180A1951D8825A2C1EE7650BFB026FDC101030000C00100000004000000881428C4A9B9F3415AC4001136DAFCC15812D1B0D150E041C81FE16D8800D9C1006098627DD6694190AAEB91B137E5C100A93375AB348641A0EA4F14BE86F8C19AF9FAB8A51FF4C10062DEF31B44ADC1343BE5C480F0D5C161963813B52100C2881428C4A9B9F3415AC4001136DAFCC15812D1B0D150E041C81FE16D8800D9C101030000C0010000000C000000881B0002E7CCF9C18E3C16AA1518E3C14E0BCFF4A15F0242DAE2E0BD6681014258245BBD8E01EBC1A89C9C59DB48DFC1D037E13D2D55D2416CC0C263C1E0E9C1A80AFE99F548FCC1BEB6FCA5CA67F7C15485041904440042203A7134FA1DF1C1B418D7FE8397F9418BD133F665CDF6C12099ABFE2353FDC176FE21475FDFF741E88C7571A905F6419224D935BF51EAC1871794B0D86CFDC1C0DFD07D1713F3C1BEC232004C02F2416002D43A5231C3419822B4807057EE4118076BBA6B45F44130BDFE632FE6EF41004A9351D275BE41C8EC31FBA8C2F9C170E79BEFD53100C2284DB8B274F2DB4100D143968EFB8F41B289F69684C2FFC157B51D885D3DF4C110F24813BFA9FC41C09512877A2AF641220C98A1D662EFC10A62C18A357CF6414A8F82D4DA76E3C1A8ED3CC40C5DDA4170A2A5B04433EE41FE9D315DBC46F241FB6CBD918D4E01C250A68D39790CF241B6C8DFD9AD8BEFC1B021F0F25B5EC7C1881B0002E7CCF9C18E3C16AA1518E3C14E0BCFF4A15F0242DAE2E0BD6681014201030000C0010000000B00000006AFEEC71EE7F5C131A8ED675FC0F4C1100D1FF28E54D3C1DCE6C866E645FCC194D2E0CD5E4CFFC10C81C0ECC127FEC110C450B315DED1C1A2E66284C2DB004200574BCF4B66D2C1109249E3DD7BF0C15E7A094BCC77F0416A6DDA9536EDF041208107445F64FB418E2B0963E5D9FBC12D00F4734CF2F1C17027C1B98FCDCF4118E03C104B82FB416CBE21E50701F7C168406B59FD96F941E2F7B3B23F1001428E2CAFEF6AA2F6416791B913CAE7F0C1F073C24A4F63014263CB6C7C1C33F3C1169DC87B75A3F8412428209BA2BEEAC16176E29845F700C2900B388F40DEF841E0DCB00D2AB6EB41DA35817C0C3CE0C174629C8A60FAED41803576F56064904178ABED5DA99FF8C164076391F9940142F8CE6AE50C0BFE4123FDA77C88CFF6C1707E6631EF9EF9C164F282B6F5E50042D03148F585A3D94117643CDBD9DBF1C106AFEEC71EE7F5C131A8ED675FC0F4C1100D1FF28E54D3C1DCE6C866E645FCC1':::GEOMETRY
					),
					(
						st_shortestline(st_scale('0104000000030000000101000000043C36F78329EFC15CE0EC676760FF410101000000980F75436D09FD41802C08695279D4C10101000000849A9F471FC8DFC188CE615F9D94DAC1':::GEOMETRY::GEOMETRY, 1.165674356848878:::FLOAT8::FLOAT8, 'NaN':::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, '0106000000020000000103000000010000000C000000295553072CC801C2C89668C808C9D2C11A6516BBB4AAEFC120747990E793CEC1DE2FC628699DE8C130129AFC0EB5DFC148090CCC7AB4D741C05F2D7D221EF0C12400CD4BEB95F041FB75F4CB8FA4F6C15EDD5BF6E6160042C908D612C77EF2C1E47B9FD0DA2DEC41808377513813A1C1C8BC849E09B8014200C4EBDE466787415050BBC65768FE41BC3A67D315150142D4FC45BBFE4CEA410CB4FB850C99ED4140BD548B0F93B941E88CC478907DFB41295553072CC801C2C89668C808C9D2C101030000000100000006000000708E027D1797C0C140009DC1CC5BE1C10C43172A53A9E841E44020AB7CEED5C1C08C23CA5E62FD41E682F7D34B16FCC138C9700D46AFE0C18C2D201238780242425AEEA0C015E0C1EAA1889F4C860142708E027D1797C0C140009DC1CC5BE1C1':::GEOMETRY::GEOMETRY)::GEOMETRY
					),
					('0101000080000000000000F87F000000000000F87F000000000000F87F':::GEOMETRY)
			)
				AS tab_24268 (col_41960)
		LIMIT
			8:::INT8;
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh sqlsmith/setup=empty/setting=no-mutations
```
</p>
</p>
</details>
/cc cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=empty/setting=no-mutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/geo/geomfn/distance.go
+++ b/pkg/geo/geomfn/distance.go
@@ -443,7 +443,7 @@ func (u *geomMinDistanceUpdater) Update(aPoint geodist.Point, bPoint geodist.Poi
 	b := bPoint.GeomPoint
 
 	dist := coordNorm(coordSub(a, b))
-	if dist < u.currentValue {
+	if dist < u.currentValue || u.coordA == nil {
 		u.currentValue = dist
 		if u.geometricalObjOrder == geometricalObjectsFlipped {
 			u.coordA = b
@@ -526,7 +526,7 @@ func (u *geomMaxDistanceUpdater) Update(aPoint geodist.Point, bPoint geodist.Poi
 	b := bPoint.GeomPoint
 
 	dist := coordNorm(coordSub(a, b))
-	if dist > u.currentValue {
+	if dist > u.currentValue || u.coordA == nil {
 		u.currentValue = dist
 		if u.geometricalObjOrder == geometricalObjectsFlipped {
 			u.coordA = b

--- a/pkg/sql/logictest/testdata/logic_test/geospatial
+++ b/pkg/sql/logictest/testdata/logic_test/geospatial
@@ -2488,6 +2488,16 @@ Square overlapping left and right square  Square (left)                         
 Square overlapping left and right square  Square (right)                            POLYGON ((0 0, -0.1 0, -0.1 1, 0 1, 0 0))
 Square overlapping left and right square  Square overlapping left and right square  POLYGON EMPTY
 
+query TT
+SELECT
+  st_astext(st_shortestline(a, b)),
+  st_astext(st_longestline(a, b))
+FROM ( VALUES
+  (st_scale('multipoint (0 0, 1 1)', 0, 'NaN'::float)::geometry, 'linestring (0 0, 1 1)'::geometry)
+) regression_t65422(a, b)
+----
+LINESTRING (0 NaN, 0 0)  LINESTRING (0 NaN, 0 0)
+
 # Binary predicates
 query TTBBBBBBBBBBBB
 SELECT

